api/v1alpha1/clientprofilemapping_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BlockPoolRefSpec indentify a block pool - client profile pair
+type BlockPoolRefSpec struct {
+	//+kubebuilder:validation:Required
+	ClientProfileName string `json:"clientProfileName,omitempty"`
```

Comment:
I disagree
This will create an issue where when we serialize with an empty value we get a valid CR (one that passes validation) but does not have a meaningful value. That is a bug.

What we need is that if we serialize with an empty value the field get removed so if you try to apply you get an error

Reply:
How is that allow, as you are adding `Required` is it allowed to create a CR with empty value?

Reply:
There is no connection between validation rules, enforced when you try to operate using the k8s API, and serialization rules, applied whenever the type is serialized into JSON or YAML. 

The intent is that whenever someone serializes the type they will not get an empty string but will get the field completely removed. So if they try to send that serialized yaml to the API server the operation will fail because the field is required. 

Also, I have a hunch (not tested yet) that the same behavior will happen if you will use the type as is and use the k8s client to send it from code. This means that if we remove `omitempty` the empty string will be serialized into an empty string on the payload which will result in k8s OpenAPI validator allowing it as a value is provided (an empty stirng)